src/build.ts

```diff
@@ -45,42 +47,41 @@ export const validateAndCreateFunctions = async (
   }
 };
 
+async function solveDenoConfigPath(
+  directory: string = Deno.cwd(),
+): Promise<string> {
+  for (const name of ["deno.json", "deno.jsonc"]) {
+    const denoConfigPath = path.join(directory, name);
+    try {
+      await Deno.stat(denoConfigPath);
+      return denoConfigPath;
+    } catch (error) {
+      if (!(error instanceof Deno.errors.NotFound)) {
+        throw error;
+      }
+    }
+  }
+  throw new Error(
+    `Could not find a deno.json or deno.jsonc file in the current directory.`,
+  );
+}
+
 const createFunctionFile = async (
+  workingDirectory: string,
   outputDirectory: string,
   fnId: string,
   fnFilePath: string,
   protocol: Protocol,
 ) => {
   const fnFileRelative = path.join("functions", `${fnId}.js`);
   const fnBundledPath = path.join(outputDirectory, fnFileRelative);
-
-  // We'll default to just using whatever Deno executable is on the path
-  // Ideally we should be able to rely on Deno.execPath() so we make sure to bundle with the same version of Deno
-  // that called this script. This is perhaps a bit overly cautious, so we can look to remove the defaulting here in the future.
-  let denoExecutablePath = "deno";
   try {
-    denoExecutablePath = Deno.execPath();
-  } catch (e) {
-    protocol.error("Error calling Deno.execPath()", e);
-  }
-
-  try {
-    // call out to deno to handle bundling
-    const p = Deno.run({
-      cmd: [
-        denoExecutablePath,
-        "bundle",
-        "--quiet",
-        fnFilePath,
-        fnBundledPath,
-      ],
+    const bundler = new EsbuildBundler({
```

Comment:
One idea for an approach here is to keep the old way of using `deno bundle`, and if the command fails, as a fallback, _then_ use `esbuild`. Might be a bit more resilient. In this way, also the default approach is the tried-and-true method of `deno bundle` and the `esbuild` fallback becomes a backup plan.